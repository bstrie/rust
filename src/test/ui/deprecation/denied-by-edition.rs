// ignore-tidy-linelength

// TODO: test error on bad edition string

#![feature(staged_api)]

#![stable(feature = "denied-by-edition-test", since = "1.0.0")]

#[rustc_deprecated(since = "1.0.0", reason = "name too short", denied_by_edition="2015")]
#[stable(feature = "denied-by-edition-test", since = "1.0.0")]
pub struct S;

fn main() {
    let _ = S; //~ ERROR use of deprecated unit struct `S`: name too short
}
