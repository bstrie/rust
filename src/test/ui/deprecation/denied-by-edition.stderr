error: use of deprecated unit struct `S`: name too short
  --> $DIR/denied-by-edition.rs:14:13
   |
LL |     let _ = S;
   |             ^
   |
   = note: `#[deny(denied_by_edition)]` on by default

error: aborting due to previous error

